cacholote/encode.py

```diff
@@ -87,11 +80,43 @@ def dictify_xarray_asset(
             pathlib.Path(config.SETTINGS["directory"]).absolute()
             / f"{checksum}{extension}"
         ),
-        "xarray:open_kwargs": open_kwargs,
-        "xarray:storage_options": storage_options,
     }
 
 
+def dictify_io_asset(
+    filetype: str,
+    checksum: str,
+    size: int,
+    extension: str = "",
+    open_kwargs: Dict[str, Any] = {},
+) -> Dict[str, Any]:
+    file_kwargs = locals()
+    extra_kwargs = {"io:open_kwargs": file_kwargs.pop("open_kwargs")}
+    return {**dictify_file(**file_kwargs), **extra_kwargs}
+
+
+def dictify_xarray_asset(
+    checksum: str,
+    size: int,
+    open_kwargs: Dict[str, Any] = {},
+    storage_options: Dict[str, Any] = {},
+) -> Dict[str, Any]:
+
+    file_kwargs = locals()
```

Comment:
I see the point, but using `locals()` is very fragile and very not-explicit. I would prefer to see the dict build explicitly.